// @tigrisdata/core/dist/proto/server/v1/billing_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

// Original file comments:
// Copyright 2022-2023 Tigris Data, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
'use strict';
var grpc = require('@grpc/grpc-js');
var server_v1_billing_pb = require('../../server/v1/billing_pb.js');
var google_api_annotations_pb = require('../../google/api/annotations_pb.js');
var openapiv3_annotations_pb = require('../../openapiv3/annotations_pb.js');
var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');

function serialize_tigrisdata_billing_v1_ListInvoicesRequest(arg) {
  if (!(arg instanceof server_v1_billing_pb.ListInvoicesRequest)) {
    throw new Error('Expected argument of type tigrisdata.billing.v1.ListInvoicesRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_billing_v1_ListInvoicesRequest(buffer_arg) {
  return server_v1_billing_pb.ListInvoicesRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_billing_v1_ListInvoicesResponse(arg) {
  if (!(arg instanceof server_v1_billing_pb.ListInvoicesResponse)) {
    throw new Error('Expected argument of type tigrisdata.billing.v1.ListInvoicesResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_billing_v1_ListInvoicesResponse(buffer_arg) {
  return server_v1_billing_pb.ListInvoicesResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var BillingService = exports.BillingService = {
  // ListInvoices fetches past invoices for this user
listInvoices: {
    path: '/tigrisdata.billing.v1.Billing/ListInvoices',
    requestStream: false,
    responseStream: false,
    requestType: server_v1_billing_pb.ListInvoicesRequest,
    responseType: server_v1_billing_pb.ListInvoicesResponse,
    requestSerialize: serialize_tigrisdata_billing_v1_ListInvoicesRequest,
    requestDeserialize: deserialize_tigrisdata_billing_v1_ListInvoicesRequest,
    responseSerialize: serialize_tigrisdata_billing_v1_ListInvoicesResponse,
    responseDeserialize: deserialize_tigrisdata_billing_v1_ListInvoicesResponse,
  },
};

exports.BillingClient = grpc.makeGenericClientConstructor(BillingService);
