// @tigrisdata/core/dist/proto/server/v1/billing_pb.js
// source: server/v1/billing.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var google_api_annotations_pb = require('../../google/api/annotations_pb.js');
goog.object.extend(proto, google_api_annotations_pb);
var openapiv3_annotations_pb = require('../../openapiv3/annotations_pb.js');
goog.object.extend(proto, openapiv3_annotations_pb);
var google_protobuf_timestamp_pb = require('google-protobuf/google/protobuf/timestamp_pb.js');
goog.object.extend(proto, google_protobuf_timestamp_pb);
goog.exportSymbol('proto.tigrisdata.billing.v1.Charge', null, global);
goog.exportSymbol('proto.tigrisdata.billing.v1.ChargeTier', null, global);
goog.exportSymbol('proto.tigrisdata.billing.v1.Invoice', null, global);
goog.exportSymbol('proto.tigrisdata.billing.v1.InvoiceLineItem', null, global);
goog.exportSymbol('proto.tigrisdata.billing.v1.ListInvoicesRequest', null, global);
goog.exportSymbol('proto.tigrisdata.billing.v1.ListInvoicesResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.billing.v1.ListInvoicesRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.billing.v1.ListInvoicesRequest.displayName = 'proto.tigrisdata.billing.v1.ListInvoicesRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.tigrisdata.billing.v1.ListInvoicesResponse.repeatedFields_, null);
};
goog.inherits(proto.tigrisdata.billing.v1.ListInvoicesResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.billing.v1.ListInvoicesResponse.displayName = 'proto.tigrisdata.billing.v1.ListInvoicesResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.billing.v1.Invoice = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.tigrisdata.billing.v1.Invoice.repeatedFields_, null);
};
goog.inherits(proto.tigrisdata.billing.v1.Invoice, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.billing.v1.Invoice.displayName = 'proto.tigrisdata.billing.v1.Invoice';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.billing.v1.InvoiceLineItem = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.tigrisdata.billing.v1.InvoiceLineItem.repeatedFields_, null);
};
goog.inherits(proto.tigrisdata.billing.v1.InvoiceLineItem, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.billing.v1.InvoiceLineItem.displayName = 'proto.tigrisdata.billing.v1.InvoiceLineItem';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.billing.v1.Charge = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.tigrisdata.billing.v1.Charge.repeatedFields_, null);
};
goog.inherits(proto.tigrisdata.billing.v1.Charge, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.billing.v1.Charge.displayName = 'proto.tigrisdata.billing.v1.Charge';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.billing.v1.ChargeTier = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.billing.v1.ChargeTier, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.billing.v1.ChargeTier.displayName = 'proto.tigrisdata.billing.v1.ChargeTier';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.billing.v1.ListInvoicesRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.billing.v1.ListInvoicesRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    startingOn: (f = msg.getStartingOn()) && google_protobuf_timestamp_pb.Timestamp.toObject(includeInstance, f),
    endingBefore: (f = msg.getEndingBefore()) && google_protobuf_timestamp_pb.Timestamp.toObject(includeInstance, f),
    invoiceId: jspb.Message.getFieldWithDefault(msg, 3, ""),
    pageSize: jspb.Message.getFieldWithDefault(msg, 4, 0),
    nextPage: jspb.Message.getFieldWithDefault(msg, 5, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.billing.v1.ListInvoicesRequest;
  return proto.tigrisdata.billing.v1.ListInvoicesRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.billing.v1.ListInvoicesRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new google_protobuf_timestamp_pb.Timestamp;
      reader.readMessage(value,google_protobuf_timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setStartingOn(value);
      break;
    case 2:
      var value = new google_protobuf_timestamp_pb.Timestamp;
      reader.readMessage(value,google_protobuf_timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setEndingBefore(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setInvoiceId(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setPageSize(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setNextPage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.billing.v1.ListInvoicesRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.billing.v1.ListInvoicesRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStartingOn();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      google_protobuf_timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getEndingBefore();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      google_protobuf_timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getInvoiceId();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = /** @type {number} */ (jspb.Message.getField(message, 4));
  if (f != null) {
    writer.writeInt32(
      4,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 5));
  if (f != null) {
    writer.writeString(
      5,
      f
    );
  }
};


/**
 * optional google.protobuf.Timestamp starting_on = 1;
 * @return {?proto.google.protobuf.Timestamp}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.getStartingOn = function() {
  return /** @type{?proto.google.protobuf.Timestamp} */ (
    jspb.Message.getWrapperField(this, google_protobuf_timestamp_pb.Timestamp, 1));
};


/**
 * @param {?proto.google.protobuf.Timestamp|undefined} value
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
*/
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.setStartingOn = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.clearStartingOn = function() {
  return this.setStartingOn(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.hasStartingOn = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * optional google.protobuf.Timestamp ending_before = 2;
 * @return {?proto.google.protobuf.Timestamp}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.getEndingBefore = function() {
  return /** @type{?proto.google.protobuf.Timestamp} */ (
    jspb.Message.getWrapperField(this, google_protobuf_timestamp_pb.Timestamp, 2));
};


/**
 * @param {?proto.google.protobuf.Timestamp|undefined} value
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
*/
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.setEndingBefore = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.clearEndingBefore = function() {
  return this.setEndingBefore(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.hasEndingBefore = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional string invoice_id = 3;
 * @return {string}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.getInvoiceId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.setInvoiceId = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional int32 page_size = 4;
 * @return {number}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.getPageSize = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.setPageSize = function(value) {
  return jspb.Message.setField(this, 4, value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.clearPageSize = function() {
  return jspb.Message.setField(this, 4, undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.hasPageSize = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional string next_page = 5;
 * @return {string}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.getNextPage = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.setNextPage = function(value) {
  return jspb.Message.setField(this, 5, value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesRequest} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.clearNextPage = function() {
  return jspb.Message.setField(this, 5, undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.billing.v1.ListInvoicesRequest.prototype.hasNextPage = function() {
  return jspb.Message.getField(this, 5) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.repeatedFields_ = [1];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.billing.v1.ListInvoicesResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.billing.v1.ListInvoicesResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    dataList: jspb.Message.toObjectList(msg.getDataList(),
    proto.tigrisdata.billing.v1.Invoice.toObject, includeInstance),
    nextPage: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesResponse}
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.billing.v1.ListInvoicesResponse;
  return proto.tigrisdata.billing.v1.ListInvoicesResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.billing.v1.ListInvoicesResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesResponse}
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.tigrisdata.billing.v1.Invoice;
      reader.readMessage(value,proto.tigrisdata.billing.v1.Invoice.deserializeBinaryFromReader);
      msg.addData(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setNextPage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.billing.v1.ListInvoicesResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.billing.v1.ListInvoicesResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDataList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      1,
      f,
      proto.tigrisdata.billing.v1.Invoice.serializeBinaryToWriter
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 2));
  if (f != null) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * repeated Invoice data = 1;
 * @return {!Array<!proto.tigrisdata.billing.v1.Invoice>}
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.getDataList = function() {
  return /** @type{!Array<!proto.tigrisdata.billing.v1.Invoice>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.tigrisdata.billing.v1.Invoice, 1));
};


/**
 * @param {!Array<!proto.tigrisdata.billing.v1.Invoice>} value
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesResponse} returns this
*/
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.setDataList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 1, value);
};


/**
 * @param {!proto.tigrisdata.billing.v1.Invoice=} opt_value
 * @param {number=} opt_index
 * @return {!proto.tigrisdata.billing.v1.Invoice}
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.addData = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 1, opt_value, proto.tigrisdata.billing.v1.Invoice, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesResponse} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.clearDataList = function() {
  return this.setDataList([]);
};


/**
 * optional string next_page = 2;
 * @return {string}
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.getNextPage = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesResponse} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.setNextPage = function(value) {
  return jspb.Message.setField(this, 2, value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.tigrisdata.billing.v1.ListInvoicesResponse} returns this
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.clearNextPage = function() {
  return jspb.Message.setField(this, 2, undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.billing.v1.ListInvoicesResponse.prototype.hasNextPage = function() {
  return jspb.Message.getField(this, 2) != null;
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.tigrisdata.billing.v1.Invoice.repeatedFields_ = [2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.billing.v1.Invoice.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.billing.v1.Invoice} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.Invoice.toObject = function(includeInstance, msg) {
  var f, obj = {
    id: jspb.Message.getFieldWithDefault(msg, 1, ""),
    entriesList: jspb.Message.toObjectList(msg.getEntriesList(),
    proto.tigrisdata.billing.v1.InvoiceLineItem.toObject, includeInstance),
    startTime: (f = msg.getStartTime()) && google_protobuf_timestamp_pb.Timestamp.toObject(includeInstance, f),
    endTime: (f = msg.getEndTime()) && google_protobuf_timestamp_pb.Timestamp.toObject(includeInstance, f),
    subtotal: jspb.Message.getFloatingPointFieldWithDefault(msg, 5, 0.0),
    total: jspb.Message.getFloatingPointFieldWithDefault(msg, 6, 0.0),
    planName: jspb.Message.getFieldWithDefault(msg, 7, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.billing.v1.Invoice}
 */
proto.tigrisdata.billing.v1.Invoice.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.billing.v1.Invoice;
  return proto.tigrisdata.billing.v1.Invoice.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.billing.v1.Invoice} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.billing.v1.Invoice}
 */
proto.tigrisdata.billing.v1.Invoice.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setId(value);
      break;
    case 2:
      var value = new proto.tigrisdata.billing.v1.InvoiceLineItem;
      reader.readMessage(value,proto.tigrisdata.billing.v1.InvoiceLineItem.deserializeBinaryFromReader);
      msg.addEntries(value);
      break;
    case 3:
      var value = new google_protobuf_timestamp_pb.Timestamp;
      reader.readMessage(value,google_protobuf_timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setStartTime(value);
      break;
    case 4:
      var value = new google_protobuf_timestamp_pb.Timestamp;
      reader.readMessage(value,google_protobuf_timestamp_pb.Timestamp.deserializeBinaryFromReader);
      msg.setEndTime(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setSubtotal(value);
      break;
    case 6:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setTotal(value);
      break;
    case 7:
      var value = /** @type {string} */ (reader.readString());
      msg.setPlanName(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.billing.v1.Invoice.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.billing.v1.Invoice} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.Invoice.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getEntriesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.tigrisdata.billing.v1.InvoiceLineItem.serializeBinaryToWriter
    );
  }
  f = message.getStartTime();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      google_protobuf_timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getEndTime();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      google_protobuf_timestamp_pb.Timestamp.serializeBinaryToWriter
    );
  }
  f = message.getSubtotal();
  if (f !== 0.0) {
    writer.writeFloat(
      5,
      f
    );
  }
  f = message.getTotal();
  if (f !== 0.0) {
    writer.writeFloat(
      6,
      f
    );
  }
  f = message.getPlanName();
  if (f.length > 0) {
    writer.writeString(
      7,
      f
    );
  }
};


/**
 * optional string id = 1;
 * @return {string}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.getId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
 */
proto.tigrisdata.billing.v1.Invoice.prototype.setId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * repeated InvoiceLineItem entries = 2;
 * @return {!Array<!proto.tigrisdata.billing.v1.InvoiceLineItem>}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.getEntriesList = function() {
  return /** @type{!Array<!proto.tigrisdata.billing.v1.InvoiceLineItem>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.tigrisdata.billing.v1.InvoiceLineItem, 2));
};


/**
 * @param {!Array<!proto.tigrisdata.billing.v1.InvoiceLineItem>} value
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
*/
proto.tigrisdata.billing.v1.Invoice.prototype.setEntriesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.tigrisdata.billing.v1.InvoiceLineItem=} opt_value
 * @param {number=} opt_index
 * @return {!proto.tigrisdata.billing.v1.InvoiceLineItem}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.addEntries = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.tigrisdata.billing.v1.InvoiceLineItem, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
 */
proto.tigrisdata.billing.v1.Invoice.prototype.clearEntriesList = function() {
  return this.setEntriesList([]);
};


/**
 * optional google.protobuf.Timestamp start_time = 3;
 * @return {?proto.google.protobuf.Timestamp}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.getStartTime = function() {
  return /** @type{?proto.google.protobuf.Timestamp} */ (
    jspb.Message.getWrapperField(this, google_protobuf_timestamp_pb.Timestamp, 3));
};


/**
 * @param {?proto.google.protobuf.Timestamp|undefined} value
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
*/
proto.tigrisdata.billing.v1.Invoice.prototype.setStartTime = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
 */
proto.tigrisdata.billing.v1.Invoice.prototype.clearStartTime = function() {
  return this.setStartTime(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.hasStartTime = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional google.protobuf.Timestamp end_time = 4;
 * @return {?proto.google.protobuf.Timestamp}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.getEndTime = function() {
  return /** @type{?proto.google.protobuf.Timestamp} */ (
    jspb.Message.getWrapperField(this, google_protobuf_timestamp_pb.Timestamp, 4));
};


/**
 * @param {?proto.google.protobuf.Timestamp|undefined} value
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
*/
proto.tigrisdata.billing.v1.Invoice.prototype.setEndTime = function(value) {
  return jspb.Message.setWrapperField(this, 4, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
 */
proto.tigrisdata.billing.v1.Invoice.prototype.clearEndTime = function() {
  return this.setEndTime(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.hasEndTime = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional float subtotal = 5;
 * @return {number}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.getSubtotal = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 5, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
 */
proto.tigrisdata.billing.v1.Invoice.prototype.setSubtotal = function(value) {
  return jspb.Message.setProto3FloatField(this, 5, value);
};


/**
 * optional float total = 6;
 * @return {number}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.getTotal = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 6, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
 */
proto.tigrisdata.billing.v1.Invoice.prototype.setTotal = function(value) {
  return jspb.Message.setProto3FloatField(this, 6, value);
};


/**
 * optional string plan_name = 7;
 * @return {string}
 */
proto.tigrisdata.billing.v1.Invoice.prototype.getPlanName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 7, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.billing.v1.Invoice} returns this
 */
proto.tigrisdata.billing.v1.Invoice.prototype.setPlanName = function(value) {
  return jspb.Message.setProto3StringField(this, 7, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.repeatedFields_ = [4];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.billing.v1.InvoiceLineItem.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.billing.v1.InvoiceLineItem} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    quantity: jspb.Message.getFloatingPointFieldWithDefault(msg, 2, 0.0),
    total: jspb.Message.getFloatingPointFieldWithDefault(msg, 3, 0.0),
    chargesList: jspb.Message.toObjectList(msg.getChargesList(),
    proto.tigrisdata.billing.v1.Charge.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.billing.v1.InvoiceLineItem}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.billing.v1.InvoiceLineItem;
  return proto.tigrisdata.billing.v1.InvoiceLineItem.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.billing.v1.InvoiceLineItem} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.billing.v1.InvoiceLineItem}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setQuantity(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setTotal(value);
      break;
    case 4:
      var value = new proto.tigrisdata.billing.v1.Charge;
      reader.readMessage(value,proto.tigrisdata.billing.v1.Charge.deserializeBinaryFromReader);
      msg.addCharges(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.billing.v1.InvoiceLineItem.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.billing.v1.InvoiceLineItem} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getQuantity();
  if (f !== 0.0) {
    writer.writeFloat(
      2,
      f
    );
  }
  f = message.getTotal();
  if (f !== 0.0) {
    writer.writeFloat(
      3,
      f
    );
  }
  f = message.getChargesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      proto.tigrisdata.billing.v1.Charge.serializeBinaryToWriter
    );
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.billing.v1.InvoiceLineItem} returns this
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional float quantity = 2;
 * @return {number}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.getQuantity = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 2, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.InvoiceLineItem} returns this
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.setQuantity = function(value) {
  return jspb.Message.setProto3FloatField(this, 2, value);
};


/**
 * optional float total = 3;
 * @return {number}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.getTotal = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 3, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.InvoiceLineItem} returns this
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.setTotal = function(value) {
  return jspb.Message.setProto3FloatField(this, 3, value);
};


/**
 * repeated Charge charges = 4;
 * @return {!Array<!proto.tigrisdata.billing.v1.Charge>}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.getChargesList = function() {
  return /** @type{!Array<!proto.tigrisdata.billing.v1.Charge>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.tigrisdata.billing.v1.Charge, 4));
};


/**
 * @param {!Array<!proto.tigrisdata.billing.v1.Charge>} value
 * @return {!proto.tigrisdata.billing.v1.InvoiceLineItem} returns this
*/
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.setChargesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.tigrisdata.billing.v1.Charge=} opt_value
 * @param {number=} opt_index
 * @return {!proto.tigrisdata.billing.v1.Charge}
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.addCharges = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.tigrisdata.billing.v1.Charge, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.tigrisdata.billing.v1.InvoiceLineItem} returns this
 */
proto.tigrisdata.billing.v1.InvoiceLineItem.prototype.clearChargesList = function() {
  return this.setChargesList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.tigrisdata.billing.v1.Charge.repeatedFields_ = [4];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.billing.v1.Charge.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.billing.v1.Charge.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.billing.v1.Charge} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.Charge.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    quantity: jspb.Message.getFloatingPointFieldWithDefault(msg, 2, 0.0),
    subtotal: jspb.Message.getFloatingPointFieldWithDefault(msg, 3, 0.0),
    tiersList: jspb.Message.toObjectList(msg.getTiersList(),
    proto.tigrisdata.billing.v1.ChargeTier.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.billing.v1.Charge}
 */
proto.tigrisdata.billing.v1.Charge.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.billing.v1.Charge;
  return proto.tigrisdata.billing.v1.Charge.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.billing.v1.Charge} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.billing.v1.Charge}
 */
proto.tigrisdata.billing.v1.Charge.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setQuantity(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setSubtotal(value);
      break;
    case 4:
      var value = new proto.tigrisdata.billing.v1.ChargeTier;
      reader.readMessage(value,proto.tigrisdata.billing.v1.ChargeTier.deserializeBinaryFromReader);
      msg.addTiers(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.billing.v1.Charge.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.billing.v1.Charge.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.billing.v1.Charge} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.Charge.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getQuantity();
  if (f !== 0.0) {
    writer.writeFloat(
      2,
      f
    );
  }
  f = message.getSubtotal();
  if (f !== 0.0) {
    writer.writeFloat(
      3,
      f
    );
  }
  f = message.getTiersList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      proto.tigrisdata.billing.v1.ChargeTier.serializeBinaryToWriter
    );
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.tigrisdata.billing.v1.Charge.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.billing.v1.Charge} returns this
 */
proto.tigrisdata.billing.v1.Charge.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional float quantity = 2;
 * @return {number}
 */
proto.tigrisdata.billing.v1.Charge.prototype.getQuantity = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 2, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.Charge} returns this
 */
proto.tigrisdata.billing.v1.Charge.prototype.setQuantity = function(value) {
  return jspb.Message.setProto3FloatField(this, 2, value);
};


/**
 * optional float subtotal = 3;
 * @return {number}
 */
proto.tigrisdata.billing.v1.Charge.prototype.getSubtotal = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 3, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.Charge} returns this
 */
proto.tigrisdata.billing.v1.Charge.prototype.setSubtotal = function(value) {
  return jspb.Message.setProto3FloatField(this, 3, value);
};


/**
 * repeated ChargeTier tiers = 4;
 * @return {!Array<!proto.tigrisdata.billing.v1.ChargeTier>}
 */
proto.tigrisdata.billing.v1.Charge.prototype.getTiersList = function() {
  return /** @type{!Array<!proto.tigrisdata.billing.v1.ChargeTier>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.tigrisdata.billing.v1.ChargeTier, 4));
};


/**
 * @param {!Array<!proto.tigrisdata.billing.v1.ChargeTier>} value
 * @return {!proto.tigrisdata.billing.v1.Charge} returns this
*/
proto.tigrisdata.billing.v1.Charge.prototype.setTiersList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.tigrisdata.billing.v1.ChargeTier=} opt_value
 * @param {number=} opt_index
 * @return {!proto.tigrisdata.billing.v1.ChargeTier}
 */
proto.tigrisdata.billing.v1.Charge.prototype.addTiers = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.tigrisdata.billing.v1.ChargeTier, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.tigrisdata.billing.v1.Charge} returns this
 */
proto.tigrisdata.billing.v1.Charge.prototype.clearTiersList = function() {
  return this.setTiersList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.billing.v1.ChargeTier.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.billing.v1.ChargeTier} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.ChargeTier.toObject = function(includeInstance, msg) {
  var f, obj = {
    startingAt: jspb.Message.getFloatingPointFieldWithDefault(msg, 1, 0.0),
    quantity: jspb.Message.getFloatingPointFieldWithDefault(msg, 2, 0.0),
    price: jspb.Message.getFloatingPointFieldWithDefault(msg, 3, 0.0),
    subtotal: jspb.Message.getFloatingPointFieldWithDefault(msg, 4, 0.0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.billing.v1.ChargeTier}
 */
proto.tigrisdata.billing.v1.ChargeTier.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.billing.v1.ChargeTier;
  return proto.tigrisdata.billing.v1.ChargeTier.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.billing.v1.ChargeTier} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.billing.v1.ChargeTier}
 */
proto.tigrisdata.billing.v1.ChargeTier.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setStartingAt(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setQuantity(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setPrice(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setSubtotal(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.billing.v1.ChargeTier.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.billing.v1.ChargeTier} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.billing.v1.ChargeTier.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getStartingAt();
  if (f !== 0.0) {
    writer.writeFloat(
      1,
      f
    );
  }
  f = message.getQuantity();
  if (f !== 0.0) {
    writer.writeFloat(
      2,
      f
    );
  }
  f = message.getPrice();
  if (f !== 0.0) {
    writer.writeFloat(
      3,
      f
    );
  }
  f = message.getSubtotal();
  if (f !== 0.0) {
    writer.writeFloat(
      4,
      f
    );
  }
};


/**
 * optional float starting_at = 1;
 * @return {number}
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.getStartingAt = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 1, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.ChargeTier} returns this
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.setStartingAt = function(value) {
  return jspb.Message.setProto3FloatField(this, 1, value);
};


/**
 * optional float quantity = 2;
 * @return {number}
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.getQuantity = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 2, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.ChargeTier} returns this
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.setQuantity = function(value) {
  return jspb.Message.setProto3FloatField(this, 2, value);
};


/**
 * optional float price = 3;
 * @return {number}
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.getPrice = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 3, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.ChargeTier} returns this
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.setPrice = function(value) {
  return jspb.Message.setProto3FloatField(this, 3, value);
};


/**
 * optional float subtotal = 4;
 * @return {number}
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.getSubtotal = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 4, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.billing.v1.ChargeTier} returns this
 */
proto.tigrisdata.billing.v1.ChargeTier.prototype.setSubtotal = function(value) {
  return jspb.Message.setProto3FloatField(this, 4, value);
};


goog.object.extend(exports, proto.tigrisdata.billing.v1);
