// @tigrisdata/core/dist/proto/server/v1/cache_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

// Original file comments:
// Copyright 2022-2023 Tigris Data, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
'use strict';
var grpc = require('@grpc/grpc-js');
var server_v1_cache_pb = require('../../server/v1/cache_pb.js');
var google_api_annotations_pb = require('../../google/api/annotations_pb.js');
var openapiv3_annotations_pb = require('../../openapiv3/annotations_pb.js');

function serialize_tigrisdata_cache_v1_CreateCacheRequest(arg) {
  if (!(arg instanceof server_v1_cache_pb.CreateCacheRequest)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.CreateCacheRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_CreateCacheRequest(buffer_arg) {
  return server_v1_cache_pb.CreateCacheRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_CreateCacheResponse(arg) {
  if (!(arg instanceof server_v1_cache_pb.CreateCacheResponse)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.CreateCacheResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_CreateCacheResponse(buffer_arg) {
  return server_v1_cache_pb.CreateCacheResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_DelRequest(arg) {
  if (!(arg instanceof server_v1_cache_pb.DelRequest)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.DelRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_DelRequest(buffer_arg) {
  return server_v1_cache_pb.DelRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_DelResponse(arg) {
  if (!(arg instanceof server_v1_cache_pb.DelResponse)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.DelResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_DelResponse(buffer_arg) {
  return server_v1_cache_pb.DelResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_DeleteCacheRequest(arg) {
  if (!(arg instanceof server_v1_cache_pb.DeleteCacheRequest)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.DeleteCacheRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_DeleteCacheRequest(buffer_arg) {
  return server_v1_cache_pb.DeleteCacheRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_DeleteCacheResponse(arg) {
  if (!(arg instanceof server_v1_cache_pb.DeleteCacheResponse)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.DeleteCacheResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_DeleteCacheResponse(buffer_arg) {
  return server_v1_cache_pb.DeleteCacheResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_GetRequest(arg) {
  if (!(arg instanceof server_v1_cache_pb.GetRequest)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.GetRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_GetRequest(buffer_arg) {
  return server_v1_cache_pb.GetRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_GetResponse(arg) {
  if (!(arg instanceof server_v1_cache_pb.GetResponse)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.GetResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_GetResponse(buffer_arg) {
  return server_v1_cache_pb.GetResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_GetSetRequest(arg) {
  if (!(arg instanceof server_v1_cache_pb.GetSetRequest)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.GetSetRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_GetSetRequest(buffer_arg) {
  return server_v1_cache_pb.GetSetRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_GetSetResponse(arg) {
  if (!(arg instanceof server_v1_cache_pb.GetSetResponse)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.GetSetResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_GetSetResponse(buffer_arg) {
  return server_v1_cache_pb.GetSetResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_KeysRequest(arg) {
  if (!(arg instanceof server_v1_cache_pb.KeysRequest)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.KeysRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_KeysRequest(buffer_arg) {
  return server_v1_cache_pb.KeysRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_KeysResponse(arg) {
  if (!(arg instanceof server_v1_cache_pb.KeysResponse)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.KeysResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_KeysResponse(buffer_arg) {
  return server_v1_cache_pb.KeysResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_ListCachesRequest(arg) {
  if (!(arg instanceof server_v1_cache_pb.ListCachesRequest)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.ListCachesRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_ListCachesRequest(buffer_arg) {
  return server_v1_cache_pb.ListCachesRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_ListCachesResponse(arg) {
  if (!(arg instanceof server_v1_cache_pb.ListCachesResponse)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.ListCachesResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_ListCachesResponse(buffer_arg) {
  return server_v1_cache_pb.ListCachesResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_SetRequest(arg) {
  if (!(arg instanceof server_v1_cache_pb.SetRequest)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.SetRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_SetRequest(buffer_arg) {
  return server_v1_cache_pb.SetRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_tigrisdata_cache_v1_SetResponse(arg) {
  if (!(arg instanceof server_v1_cache_pb.SetResponse)) {
    throw new Error('Expected argument of type tigrisdata.cache.v1.SetResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_tigrisdata_cache_v1_SetResponse(buffer_arg) {
  return server_v1_cache_pb.SetResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var CacheService = exports.CacheService = {
  createCache: {
    path: '/tigrisdata.cache.v1.Cache/CreateCache',
    requestStream: false,
    responseStream: false,
    requestType: server_v1_cache_pb.CreateCacheRequest,
    responseType: server_v1_cache_pb.CreateCacheResponse,
    requestSerialize: serialize_tigrisdata_cache_v1_CreateCacheRequest,
    requestDeserialize: deserialize_tigrisdata_cache_v1_CreateCacheRequest,
    responseSerialize: serialize_tigrisdata_cache_v1_CreateCacheResponse,
    responseDeserialize: deserialize_tigrisdata_cache_v1_CreateCacheResponse,
  },
  listCaches: {
    path: '/tigrisdata.cache.v1.Cache/ListCaches',
    requestStream: false,
    responseStream: false,
    requestType: server_v1_cache_pb.ListCachesRequest,
    responseType: server_v1_cache_pb.ListCachesResponse,
    requestSerialize: serialize_tigrisdata_cache_v1_ListCachesRequest,
    requestDeserialize: deserialize_tigrisdata_cache_v1_ListCachesRequest,
    responseSerialize: serialize_tigrisdata_cache_v1_ListCachesResponse,
    responseDeserialize: deserialize_tigrisdata_cache_v1_ListCachesResponse,
  },
  deleteCache: {
    path: '/tigrisdata.cache.v1.Cache/DeleteCache',
    requestStream: false,
    responseStream: false,
    requestType: server_v1_cache_pb.DeleteCacheRequest,
    responseType: server_v1_cache_pb.DeleteCacheResponse,
    requestSerialize: serialize_tigrisdata_cache_v1_DeleteCacheRequest,
    requestDeserialize: deserialize_tigrisdata_cache_v1_DeleteCacheRequest,
    responseSerialize: serialize_tigrisdata_cache_v1_DeleteCacheResponse,
    responseDeserialize: deserialize_tigrisdata_cache_v1_DeleteCacheResponse,
  },
  set: {
    path: '/tigrisdata.cache.v1.Cache/Set',
    requestStream: false,
    responseStream: false,
    requestType: server_v1_cache_pb.SetRequest,
    responseType: server_v1_cache_pb.SetResponse,
    requestSerialize: serialize_tigrisdata_cache_v1_SetRequest,
    requestDeserialize: deserialize_tigrisdata_cache_v1_SetRequest,
    responseSerialize: serialize_tigrisdata_cache_v1_SetResponse,
    responseDeserialize: deserialize_tigrisdata_cache_v1_SetResponse,
  },
  getSet: {
    path: '/tigrisdata.cache.v1.Cache/GetSet',
    requestStream: false,
    responseStream: false,
    requestType: server_v1_cache_pb.GetSetRequest,
    responseType: server_v1_cache_pb.GetSetResponse,
    requestSerialize: serialize_tigrisdata_cache_v1_GetSetRequest,
    requestDeserialize: deserialize_tigrisdata_cache_v1_GetSetRequest,
    responseSerialize: serialize_tigrisdata_cache_v1_GetSetResponse,
    responseDeserialize: deserialize_tigrisdata_cache_v1_GetSetResponse,
  },
  get: {
    path: '/tigrisdata.cache.v1.Cache/Get',
    requestStream: false,
    responseStream: false,
    requestType: server_v1_cache_pb.GetRequest,
    responseType: server_v1_cache_pb.GetResponse,
    requestSerialize: serialize_tigrisdata_cache_v1_GetRequest,
    requestDeserialize: deserialize_tigrisdata_cache_v1_GetRequest,
    responseSerialize: serialize_tigrisdata_cache_v1_GetResponse,
    responseDeserialize: deserialize_tigrisdata_cache_v1_GetResponse,
  },
  del: {
    path: '/tigrisdata.cache.v1.Cache/Del',
    requestStream: false,
    responseStream: false,
    requestType: server_v1_cache_pb.DelRequest,
    responseType: server_v1_cache_pb.DelResponse,
    requestSerialize: serialize_tigrisdata_cache_v1_DelRequest,
    requestDeserialize: deserialize_tigrisdata_cache_v1_DelRequest,
    responseSerialize: serialize_tigrisdata_cache_v1_DelResponse,
    responseDeserialize: deserialize_tigrisdata_cache_v1_DelResponse,
  },
  keys: {
    path: '/tigrisdata.cache.v1.Cache/Keys',
    requestStream: false,
    responseStream: true,
    requestType: server_v1_cache_pb.KeysRequest,
    responseType: server_v1_cache_pb.KeysResponse,
    requestSerialize: serialize_tigrisdata_cache_v1_KeysRequest,
    requestDeserialize: deserialize_tigrisdata_cache_v1_KeysRequest,
    responseSerialize: serialize_tigrisdata_cache_v1_KeysResponse,
    responseDeserialize: deserialize_tigrisdata_cache_v1_KeysResponse,
  },
};

exports.CacheClient = grpc.makeGenericClientConstructor(CacheService);
