// @tigrisdata/core/dist/proto/server/v1/observability_pb.js
// source: server/v1/observability.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var google_api_annotations_pb = require('../../google/api/annotations_pb.js');
goog.object.extend(proto, google_api_annotations_pb);
var openapiv3_annotations_pb = require('../../openapiv3/annotations_pb.js');
goog.object.extend(proto, openapiv3_annotations_pb);
goog.exportSymbol('proto.tigrisdata.observability.v1.AdditionalFunction', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.Code', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.DataPoint', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.Error', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.ErrorDetails', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.GetInfoRequest', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.GetInfoResponse', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.MetricQueryFunction', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.MetricQuerySpaceAggregation', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.MetricSeries', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.QuotaLimitsRequest', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.QuotaLimitsResponse', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.QuotaUsageRequest', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.QuotaUsageResponse', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.RetryInfo', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.RollupAggregator', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.RollupFunction', null, global);
goog.exportSymbol('proto.tigrisdata.observability.v1.TigrisOperation', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.RetryInfo = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.RetryInfo, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.RetryInfo.displayName = 'proto.tigrisdata.observability.v1.RetryInfo';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.ErrorDetails = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.ErrorDetails, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.ErrorDetails.displayName = 'proto.tigrisdata.observability.v1.ErrorDetails';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.Error = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.Error, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.Error.displayName = 'proto.tigrisdata.observability.v1.Error';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.repeatedFields_, null);
};
goog.inherits(proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.displayName = 'proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.AdditionalFunction = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.AdditionalFunction, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.AdditionalFunction.displayName = 'proto.tigrisdata.observability.v1.AdditionalFunction';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.RollupFunction = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.RollupFunction, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.RollupFunction.displayName = 'proto.tigrisdata.observability.v1.RollupFunction';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.repeatedFields_, null);
};
goog.inherits(proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.displayName = 'proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.MetricSeries = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.tigrisdata.observability.v1.MetricSeries.repeatedFields_, null);
};
goog.inherits(proto.tigrisdata.observability.v1.MetricSeries, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.MetricSeries.displayName = 'proto.tigrisdata.observability.v1.MetricSeries';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.DataPoint = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.DataPoint, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.DataPoint.displayName = 'proto.tigrisdata.observability.v1.DataPoint';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.QuotaLimitsRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.QuotaLimitsRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.QuotaLimitsRequest.displayName = 'proto.tigrisdata.observability.v1.QuotaLimitsRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.QuotaLimitsResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.QuotaLimitsResponse.displayName = 'proto.tigrisdata.observability.v1.QuotaLimitsResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.QuotaUsageRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.QuotaUsageRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.QuotaUsageRequest.displayName = 'proto.tigrisdata.observability.v1.QuotaUsageRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.QuotaUsageResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.QuotaUsageResponse.displayName = 'proto.tigrisdata.observability.v1.QuotaUsageResponse';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.GetInfoRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.GetInfoRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.GetInfoRequest.displayName = 'proto.tigrisdata.observability.v1.GetInfoRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.tigrisdata.observability.v1.GetInfoResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.tigrisdata.observability.v1.GetInfoResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.tigrisdata.observability.v1.GetInfoResponse.displayName = 'proto.tigrisdata.observability.v1.GetInfoResponse';
}



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.RetryInfo.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.RetryInfo.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.RetryInfo} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.RetryInfo.toObject = function(includeInstance, msg) {
  var f, obj = {
    delay: jspb.Message.getFieldWithDefault(msg, 3, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.RetryInfo}
 */
proto.tigrisdata.observability.v1.RetryInfo.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.RetryInfo;
  return proto.tigrisdata.observability.v1.RetryInfo.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.RetryInfo} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.RetryInfo}
 */
proto.tigrisdata.observability.v1.RetryInfo.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 3:
      var value = /** @type {number} */ (reader.readInt32());
      msg.setDelay(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.RetryInfo.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.RetryInfo.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.RetryInfo} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.RetryInfo.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDelay();
  if (f !== 0) {
    writer.writeInt32(
      3,
      f
    );
  }
};


/**
 * optional int32 delay = 3;
 * @return {number}
 */
proto.tigrisdata.observability.v1.RetryInfo.prototype.getDelay = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.RetryInfo} returns this
 */
proto.tigrisdata.observability.v1.RetryInfo.prototype.setDelay = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.ErrorDetails.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.ErrorDetails} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.ErrorDetails.toObject = function(includeInstance, msg) {
  var f, obj = {
    code: jspb.Message.getFieldWithDefault(msg, 1, ""),
    message: jspb.Message.getFieldWithDefault(msg, 2, ""),
    retry: (f = msg.getRetry()) && proto.tigrisdata.observability.v1.RetryInfo.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.ErrorDetails}
 */
proto.tigrisdata.observability.v1.ErrorDetails.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.ErrorDetails;
  return proto.tigrisdata.observability.v1.ErrorDetails.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.ErrorDetails} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.ErrorDetails}
 */
proto.tigrisdata.observability.v1.ErrorDetails.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setCode(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setMessage(value);
      break;
    case 3:
      var value = new proto.tigrisdata.observability.v1.RetryInfo;
      reader.readMessage(value,proto.tigrisdata.observability.v1.RetryInfo.deserializeBinaryFromReader);
      msg.setRetry(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.ErrorDetails.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.ErrorDetails} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.ErrorDetails.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCode();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getMessage();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getRetry();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      proto.tigrisdata.observability.v1.RetryInfo.serializeBinaryToWriter
    );
  }
};


/**
 * optional string code = 1;
 * @return {string}
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.getCode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.ErrorDetails} returns this
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.setCode = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string message = 2;
 * @return {string}
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.getMessage = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.ErrorDetails} returns this
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.setMessage = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional RetryInfo retry = 3;
 * @return {?proto.tigrisdata.observability.v1.RetryInfo}
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.getRetry = function() {
  return /** @type{?proto.tigrisdata.observability.v1.RetryInfo} */ (
    jspb.Message.getWrapperField(this, proto.tigrisdata.observability.v1.RetryInfo, 3));
};


/**
 * @param {?proto.tigrisdata.observability.v1.RetryInfo|undefined} value
 * @return {!proto.tigrisdata.observability.v1.ErrorDetails} returns this
*/
proto.tigrisdata.observability.v1.ErrorDetails.prototype.setRetry = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.tigrisdata.observability.v1.ErrorDetails} returns this
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.clearRetry = function() {
  return this.setRetry(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.observability.v1.ErrorDetails.prototype.hasRetry = function() {
  return jspb.Message.getField(this, 3) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.Error.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.Error.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.Error} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.Error.toObject = function(includeInstance, msg) {
  var f, obj = {
    code: jspb.Message.getFieldWithDefault(msg, 1, 0),
    message: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.Error}
 */
proto.tigrisdata.observability.v1.Error.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.Error;
  return proto.tigrisdata.observability.v1.Error.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.Error} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.Error}
 */
proto.tigrisdata.observability.v1.Error.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.tigrisdata.observability.v1.Code} */ (reader.readEnum());
      msg.setCode(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setMessage(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.Error.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.Error.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.Error} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.Error.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getCode();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getMessage();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional Code code = 1;
 * @return {!proto.tigrisdata.observability.v1.Code}
 */
proto.tigrisdata.observability.v1.Error.prototype.getCode = function() {
  return /** @type {!proto.tigrisdata.observability.v1.Code} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.tigrisdata.observability.v1.Code} value
 * @return {!proto.tigrisdata.observability.v1.Error} returns this
 */
proto.tigrisdata.observability.v1.Error.prototype.setCode = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional string message = 2;
 * @return {string}
 */
proto.tigrisdata.observability.v1.Error.prototype.getMessage = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.Error} returns this
 */
proto.tigrisdata.observability.v1.Error.prototype.setMessage = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.repeatedFields_ = [9,12];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    db: jspb.Message.getFieldWithDefault(msg, 1, ""),
    branch: jspb.Message.getFieldWithDefault(msg, 2, ""),
    collection: jspb.Message.getFieldWithDefault(msg, 3, ""),
    from: jspb.Message.getFieldWithDefault(msg, 4, 0),
    to: jspb.Message.getFieldWithDefault(msg, 5, 0),
    metricName: jspb.Message.getFieldWithDefault(msg, 6, ""),
    tigrisOperation: jspb.Message.getFieldWithDefault(msg, 7, 0),
    spaceAggregation: jspb.Message.getFieldWithDefault(msg, 8, 0),
    spaceAggregatedByList: (f = jspb.Message.getRepeatedField(msg, 9)) == null ? undefined : f,
    pb_function: jspb.Message.getFieldWithDefault(msg, 10, 0),
    quantile: jspb.Message.getFloatingPointFieldWithDefault(msg, 11, 0.0),
    additionalfunctionsList: jspb.Message.toObjectList(msg.getAdditionalfunctionsList(),
    proto.tigrisdata.observability.v1.AdditionalFunction.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest;
  return proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setDb(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setBranch(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setCollection(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setFrom(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setTo(value);
      break;
    case 6:
      var value = /** @type {string} */ (reader.readString());
      msg.setMetricName(value);
      break;
    case 7:
      var value = /** @type {!proto.tigrisdata.observability.v1.TigrisOperation} */ (reader.readEnum());
      msg.setTigrisOperation(value);
      break;
    case 8:
      var value = /** @type {!proto.tigrisdata.observability.v1.MetricQuerySpaceAggregation} */ (reader.readEnum());
      msg.setSpaceAggregation(value);
      break;
    case 9:
      var value = /** @type {string} */ (reader.readString());
      msg.addSpaceAggregatedBy(value);
      break;
    case 10:
      var value = /** @type {!proto.tigrisdata.observability.v1.MetricQueryFunction} */ (reader.readEnum());
      msg.setFunction(value);
      break;
    case 11:
      var value = /** @type {number} */ (reader.readFloat());
      msg.setQuantile(value);
      break;
    case 12:
      var value = new proto.tigrisdata.observability.v1.AdditionalFunction;
      reader.readMessage(value,proto.tigrisdata.observability.v1.AdditionalFunction.deserializeBinaryFromReader);
      msg.addAdditionalfunctions(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getDb();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getBranch();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getCollection();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getFrom();
  if (f !== 0) {
    writer.writeInt64(
      4,
      f
    );
  }
  f = message.getTo();
  if (f !== 0) {
    writer.writeInt64(
      5,
      f
    );
  }
  f = message.getMetricName();
  if (f.length > 0) {
    writer.writeString(
      6,
      f
    );
  }
  f = message.getTigrisOperation();
  if (f !== 0.0) {
    writer.writeEnum(
      7,
      f
    );
  }
  f = message.getSpaceAggregation();
  if (f !== 0.0) {
    writer.writeEnum(
      8,
      f
    );
  }
  f = message.getSpaceAggregatedByList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      9,
      f
    );
  }
  f = message.getFunction();
  if (f !== 0.0) {
    writer.writeEnum(
      10,
      f
    );
  }
  f = message.getQuantile();
  if (f !== 0.0) {
    writer.writeFloat(
      11,
      f
    );
  }
  f = message.getAdditionalfunctionsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      12,
      f,
      proto.tigrisdata.observability.v1.AdditionalFunction.serializeBinaryToWriter
    );
  }
};


/**
 * optional string db = 1;
 * @return {string}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getDb = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setDb = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string branch = 2;
 * @return {string}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getBranch = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setBranch = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string collection = 3;
 * @return {string}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getCollection = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setCollection = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional int64 from = 4;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getFrom = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setFrom = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional int64 to = 5;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getTo = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setTo = function(value) {
  return jspb.Message.setProto3IntField(this, 5, value);
};


/**
 * optional string metric_name = 6;
 * @return {string}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getMetricName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setMetricName = function(value) {
  return jspb.Message.setProto3StringField(this, 6, value);
};


/**
 * optional TigrisOperation tigris_operation = 7;
 * @return {!proto.tigrisdata.observability.v1.TigrisOperation}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getTigrisOperation = function() {
  return /** @type {!proto.tigrisdata.observability.v1.TigrisOperation} */ (jspb.Message.getFieldWithDefault(this, 7, 0));
};


/**
 * @param {!proto.tigrisdata.observability.v1.TigrisOperation} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setTigrisOperation = function(value) {
  return jspb.Message.setProto3EnumField(this, 7, value);
};


/**
 * optional MetricQuerySpaceAggregation space_aggregation = 8;
 * @return {!proto.tigrisdata.observability.v1.MetricQuerySpaceAggregation}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getSpaceAggregation = function() {
  return /** @type {!proto.tigrisdata.observability.v1.MetricQuerySpaceAggregation} */ (jspb.Message.getFieldWithDefault(this, 8, 0));
};


/**
 * @param {!proto.tigrisdata.observability.v1.MetricQuerySpaceAggregation} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setSpaceAggregation = function(value) {
  return jspb.Message.setProto3EnumField(this, 8, value);
};


/**
 * repeated string space_aggregated_by = 9;
 * @return {!Array<string>}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getSpaceAggregatedByList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 9));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setSpaceAggregatedByList = function(value) {
  return jspb.Message.setField(this, 9, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.addSpaceAggregatedBy = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 9, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.clearSpaceAggregatedByList = function() {
  return this.setSpaceAggregatedByList([]);
};


/**
 * optional MetricQueryFunction function = 10;
 * @return {!proto.tigrisdata.observability.v1.MetricQueryFunction}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getFunction = function() {
  return /** @type {!proto.tigrisdata.observability.v1.MetricQueryFunction} */ (jspb.Message.getFieldWithDefault(this, 10, 0));
};


/**
 * @param {!proto.tigrisdata.observability.v1.MetricQueryFunction} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setFunction = function(value) {
  return jspb.Message.setProto3EnumField(this, 10, value);
};


/**
 * optional float quantile = 11;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getQuantile = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 11, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setQuantile = function(value) {
  return jspb.Message.setProto3FloatField(this, 11, value);
};


/**
 * repeated AdditionalFunction additionalFunctions = 12;
 * @return {!Array<!proto.tigrisdata.observability.v1.AdditionalFunction>}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.getAdditionalfunctionsList = function() {
  return /** @type{!Array<!proto.tigrisdata.observability.v1.AdditionalFunction>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.tigrisdata.observability.v1.AdditionalFunction, 12));
};


/**
 * @param {!Array<!proto.tigrisdata.observability.v1.AdditionalFunction>} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
*/
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.setAdditionalfunctionsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 12, value);
};


/**
 * @param {!proto.tigrisdata.observability.v1.AdditionalFunction=} opt_value
 * @param {number=} opt_index
 * @return {!proto.tigrisdata.observability.v1.AdditionalFunction}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.addAdditionalfunctions = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 12, opt_value, proto.tigrisdata.observability.v1.AdditionalFunction, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsRequest.prototype.clearAdditionalfunctionsList = function() {
  return this.setAdditionalfunctionsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.AdditionalFunction.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.AdditionalFunction.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.AdditionalFunction} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.AdditionalFunction.toObject = function(includeInstance, msg) {
  var f, obj = {
    rollup: (f = msg.getRollup()) && proto.tigrisdata.observability.v1.RollupFunction.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.AdditionalFunction}
 */
proto.tigrisdata.observability.v1.AdditionalFunction.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.AdditionalFunction;
  return proto.tigrisdata.observability.v1.AdditionalFunction.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.AdditionalFunction} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.AdditionalFunction}
 */
proto.tigrisdata.observability.v1.AdditionalFunction.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = new proto.tigrisdata.observability.v1.RollupFunction;
      reader.readMessage(value,proto.tigrisdata.observability.v1.RollupFunction.deserializeBinaryFromReader);
      msg.setRollup(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.AdditionalFunction.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.AdditionalFunction.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.AdditionalFunction} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.AdditionalFunction.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getRollup();
  if (f != null) {
    writer.writeMessage(
      1,
      f,
      proto.tigrisdata.observability.v1.RollupFunction.serializeBinaryToWriter
    );
  }
};


/**
 * optional RollupFunction rollup = 1;
 * @return {?proto.tigrisdata.observability.v1.RollupFunction}
 */
proto.tigrisdata.observability.v1.AdditionalFunction.prototype.getRollup = function() {
  return /** @type{?proto.tigrisdata.observability.v1.RollupFunction} */ (
    jspb.Message.getWrapperField(this, proto.tigrisdata.observability.v1.RollupFunction, 1));
};


/**
 * @param {?proto.tigrisdata.observability.v1.RollupFunction|undefined} value
 * @return {!proto.tigrisdata.observability.v1.AdditionalFunction} returns this
*/
proto.tigrisdata.observability.v1.AdditionalFunction.prototype.setRollup = function(value) {
  return jspb.Message.setWrapperField(this, 1, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.tigrisdata.observability.v1.AdditionalFunction} returns this
 */
proto.tigrisdata.observability.v1.AdditionalFunction.prototype.clearRollup = function() {
  return this.setRollup(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.observability.v1.AdditionalFunction.prototype.hasRollup = function() {
  return jspb.Message.getField(this, 1) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.RollupFunction.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.RollupFunction.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.RollupFunction} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.RollupFunction.toObject = function(includeInstance, msg) {
  var f, obj = {
    aggregator: jspb.Message.getFieldWithDefault(msg, 1, 0),
    interval: jspb.Message.getFieldWithDefault(msg, 2, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.RollupFunction}
 */
proto.tigrisdata.observability.v1.RollupFunction.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.RollupFunction;
  return proto.tigrisdata.observability.v1.RollupFunction.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.RollupFunction} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.RollupFunction}
 */
proto.tigrisdata.observability.v1.RollupFunction.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {!proto.tigrisdata.observability.v1.RollupAggregator} */ (reader.readEnum());
      msg.setAggregator(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setInterval(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.RollupFunction.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.RollupFunction.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.RollupFunction} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.RollupFunction.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getAggregator();
  if (f !== 0.0) {
    writer.writeEnum(
      1,
      f
    );
  }
  f = message.getInterval();
  if (f !== 0) {
    writer.writeInt64(
      2,
      f
    );
  }
};


/**
 * optional RollupAggregator aggregator = 1;
 * @return {!proto.tigrisdata.observability.v1.RollupAggregator}
 */
proto.tigrisdata.observability.v1.RollupFunction.prototype.getAggregator = function() {
  return /** @type {!proto.tigrisdata.observability.v1.RollupAggregator} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {!proto.tigrisdata.observability.v1.RollupAggregator} value
 * @return {!proto.tigrisdata.observability.v1.RollupFunction} returns this
 */
proto.tigrisdata.observability.v1.RollupFunction.prototype.setAggregator = function(value) {
  return jspb.Message.setProto3EnumField(this, 1, value);
};


/**
 * optional int64 interval = 2;
 * @return {number}
 */
proto.tigrisdata.observability.v1.RollupFunction.prototype.getInterval = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.RollupFunction} returns this
 */
proto.tigrisdata.observability.v1.RollupFunction.prototype.setInterval = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.repeatedFields_ = [4];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    from: jspb.Message.getFieldWithDefault(msg, 1, 0),
    to: jspb.Message.getFieldWithDefault(msg, 2, 0),
    query: jspb.Message.getFieldWithDefault(msg, 3, ""),
    seriesList: jspb.Message.toObjectList(msg.getSeriesList(),
    proto.tigrisdata.observability.v1.MetricSeries.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse;
  return proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setFrom(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setTo(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setQuery(value);
      break;
    case 4:
      var value = new proto.tigrisdata.observability.v1.MetricSeries;
      reader.readMessage(value,proto.tigrisdata.observability.v1.MetricSeries.deserializeBinaryFromReader);
      msg.addSeries(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getFrom();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getTo();
  if (f !== 0) {
    writer.writeInt64(
      2,
      f
    );
  }
  f = message.getQuery();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getSeriesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      4,
      f,
      proto.tigrisdata.observability.v1.MetricSeries.serializeBinaryToWriter
    );
  }
};


/**
 * optional int64 from = 1;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.getFrom = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.setFrom = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional int64 to = 2;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.getTo = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.setTo = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional string query = 3;
 * @return {string}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.getQuery = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.setQuery = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * repeated MetricSeries series = 4;
 * @return {!Array<!proto.tigrisdata.observability.v1.MetricSeries>}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.getSeriesList = function() {
  return /** @type{!Array<!proto.tigrisdata.observability.v1.MetricSeries>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.tigrisdata.observability.v1.MetricSeries, 4));
};


/**
 * @param {!Array<!proto.tigrisdata.observability.v1.MetricSeries>} value
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse} returns this
*/
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.setSeriesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 4, value);
};


/**
 * @param {!proto.tigrisdata.observability.v1.MetricSeries=} opt_value
 * @param {number=} opt_index
 * @return {!proto.tigrisdata.observability.v1.MetricSeries}
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.addSeries = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 4, opt_value, proto.tigrisdata.observability.v1.MetricSeries, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse} returns this
 */
proto.tigrisdata.observability.v1.QueryTimeSeriesMetricsResponse.prototype.clearSeriesList = function() {
  return this.setSeriesList([]);
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.tigrisdata.observability.v1.MetricSeries.repeatedFields_ = [5];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.MetricSeries.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.MetricSeries} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.MetricSeries.toObject = function(includeInstance, msg) {
  var f, obj = {
    from: jspb.Message.getFieldWithDefault(msg, 1, 0),
    to: jspb.Message.getFieldWithDefault(msg, 2, 0),
    metric: jspb.Message.getFieldWithDefault(msg, 3, ""),
    scope: jspb.Message.getFieldWithDefault(msg, 4, ""),
    datapointsList: jspb.Message.toObjectList(msg.getDatapointsList(),
    proto.tigrisdata.observability.v1.DataPoint.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.MetricSeries}
 */
proto.tigrisdata.observability.v1.MetricSeries.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.MetricSeries;
  return proto.tigrisdata.observability.v1.MetricSeries.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.MetricSeries} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.MetricSeries}
 */
proto.tigrisdata.observability.v1.MetricSeries.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setFrom(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setTo(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setMetric(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setScope(value);
      break;
    case 5:
      var value = new proto.tigrisdata.observability.v1.DataPoint;
      reader.readMessage(value,proto.tigrisdata.observability.v1.DataPoint.deserializeBinaryFromReader);
      msg.addDatapoints(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.MetricSeries.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.MetricSeries} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.MetricSeries.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getFrom();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getTo();
  if (f !== 0) {
    writer.writeInt64(
      2,
      f
    );
  }
  f = message.getMetric();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getScope();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getDatapointsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      proto.tigrisdata.observability.v1.DataPoint.serializeBinaryToWriter
    );
  }
};


/**
 * optional int64 from = 1;
 * @return {number}
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.getFrom = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.MetricSeries} returns this
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.setFrom = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional int64 to = 2;
 * @return {number}
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.getTo = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.MetricSeries} returns this
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.setTo = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional string metric = 3;
 * @return {string}
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.getMetric = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.MetricSeries} returns this
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.setMetric = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string scope = 4;
 * @return {string}
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.getScope = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.MetricSeries} returns this
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.setScope = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * repeated DataPoint dataPoints = 5;
 * @return {!Array<!proto.tigrisdata.observability.v1.DataPoint>}
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.getDatapointsList = function() {
  return /** @type{!Array<!proto.tigrisdata.observability.v1.DataPoint>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.tigrisdata.observability.v1.DataPoint, 5));
};


/**
 * @param {!Array<!proto.tigrisdata.observability.v1.DataPoint>} value
 * @return {!proto.tigrisdata.observability.v1.MetricSeries} returns this
*/
proto.tigrisdata.observability.v1.MetricSeries.prototype.setDatapointsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.tigrisdata.observability.v1.DataPoint=} opt_value
 * @param {number=} opt_index
 * @return {!proto.tigrisdata.observability.v1.DataPoint}
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.addDatapoints = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.tigrisdata.observability.v1.DataPoint, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.tigrisdata.observability.v1.MetricSeries} returns this
 */
proto.tigrisdata.observability.v1.MetricSeries.prototype.clearDatapointsList = function() {
  return this.setDatapointsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.DataPoint.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.DataPoint.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.DataPoint} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.DataPoint.toObject = function(includeInstance, msg) {
  var f, obj = {
    timestamp: jspb.Message.getFieldWithDefault(msg, 1, 0),
    value: jspb.Message.getFloatingPointFieldWithDefault(msg, 2, 0.0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.DataPoint}
 */
proto.tigrisdata.observability.v1.DataPoint.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.DataPoint;
  return proto.tigrisdata.observability.v1.DataPoint.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.DataPoint} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.DataPoint}
 */
proto.tigrisdata.observability.v1.DataPoint.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setTimestamp(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readDouble());
      msg.setValue(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.DataPoint.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.DataPoint.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.DataPoint} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.DataPoint.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getTimestamp();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getValue();
  if (f !== 0.0) {
    writer.writeDouble(
      2,
      f
    );
  }
};


/**
 * optional int64 timestamp = 1;
 * @return {number}
 */
proto.tigrisdata.observability.v1.DataPoint.prototype.getTimestamp = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.DataPoint} returns this
 */
proto.tigrisdata.observability.v1.DataPoint.prototype.setTimestamp = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional double value = 2;
 * @return {number}
 */
proto.tigrisdata.observability.v1.DataPoint.prototype.getValue = function() {
  return /** @type {number} */ (jspb.Message.getFloatingPointFieldWithDefault(this, 2, 0.0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.DataPoint} returns this
 */
proto.tigrisdata.observability.v1.DataPoint.prototype.setValue = function(value) {
  return jspb.Message.setProto3FloatField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.QuotaLimitsRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.QuotaLimitsRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.QuotaLimitsRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QuotaLimitsRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.QuotaLimitsRequest}
 */
proto.tigrisdata.observability.v1.QuotaLimitsRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.QuotaLimitsRequest;
  return proto.tigrisdata.observability.v1.QuotaLimitsRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.QuotaLimitsRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.QuotaLimitsRequest}
 */
proto.tigrisdata.observability.v1.QuotaLimitsRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.QuotaLimitsRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.QuotaLimitsRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.QuotaLimitsRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QuotaLimitsRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.QuotaLimitsResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.QuotaLimitsResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    readunits: jspb.Message.getFieldWithDefault(msg, 1, 0),
    writeunits: jspb.Message.getFieldWithDefault(msg, 2, 0),
    storagesize: jspb.Message.getFieldWithDefault(msg, 3, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.QuotaLimitsResponse}
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.QuotaLimitsResponse;
  return proto.tigrisdata.observability.v1.QuotaLimitsResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.QuotaLimitsResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.QuotaLimitsResponse}
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setReadunits(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setWriteunits(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setStoragesize(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.QuotaLimitsResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.QuotaLimitsResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getReadunits();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getWriteunits();
  if (f !== 0) {
    writer.writeInt64(
      2,
      f
    );
  }
  f = message.getStoragesize();
  if (f !== 0) {
    writer.writeInt64(
      3,
      f
    );
  }
};


/**
 * optional int64 ReadUnits = 1;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.prototype.getReadunits = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaLimitsResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.prototype.setReadunits = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional int64 WriteUnits = 2;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.prototype.getWriteunits = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaLimitsResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.prototype.setWriteunits = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional int64 StorageSize = 3;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.prototype.getStoragesize = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaLimitsResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaLimitsResponse.prototype.setStoragesize = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.QuotaUsageRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.QuotaUsageRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.QuotaUsageRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QuotaUsageRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageRequest}
 */
proto.tigrisdata.observability.v1.QuotaUsageRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.QuotaUsageRequest;
  return proto.tigrisdata.observability.v1.QuotaUsageRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.QuotaUsageRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageRequest}
 */
proto.tigrisdata.observability.v1.QuotaUsageRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.QuotaUsageRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.QuotaUsageRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.QuotaUsageRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QuotaUsageRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.QuotaUsageResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.QuotaUsageResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    readunits: jspb.Message.getFieldWithDefault(msg, 1, 0),
    writeunits: jspb.Message.getFieldWithDefault(msg, 2, 0),
    storagesize: jspb.Message.getFieldWithDefault(msg, 3, 0),
    readunitsthrottled: jspb.Message.getFieldWithDefault(msg, 4, 0),
    writeunitsthrottled: jspb.Message.getFieldWithDefault(msg, 5, 0),
    storagesizethrottled: jspb.Message.getFieldWithDefault(msg, 6, 0)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageResponse}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.QuotaUsageResponse;
  return proto.tigrisdata.observability.v1.QuotaUsageResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.QuotaUsageResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageResponse}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setReadunits(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setWriteunits(value);
      break;
    case 3:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setStoragesize(value);
      break;
    case 4:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setReadunitsthrottled(value);
      break;
    case 5:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setWriteunitsthrottled(value);
      break;
    case 6:
      var value = /** @type {number} */ (reader.readInt64());
      msg.setStoragesizethrottled(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.QuotaUsageResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.QuotaUsageResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getReadunits();
  if (f !== 0) {
    writer.writeInt64(
      1,
      f
    );
  }
  f = message.getWriteunits();
  if (f !== 0) {
    writer.writeInt64(
      2,
      f
    );
  }
  f = message.getStoragesize();
  if (f !== 0) {
    writer.writeInt64(
      3,
      f
    );
  }
  f = message.getReadunitsthrottled();
  if (f !== 0) {
    writer.writeInt64(
      4,
      f
    );
  }
  f = message.getWriteunitsthrottled();
  if (f !== 0) {
    writer.writeInt64(
      5,
      f
    );
  }
  f = message.getStoragesizethrottled();
  if (f !== 0) {
    writer.writeInt64(
      6,
      f
    );
  }
};


/**
 * optional int64 ReadUnits = 1;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.getReadunits = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 1, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.setReadunits = function(value) {
  return jspb.Message.setProto3IntField(this, 1, value);
};


/**
 * optional int64 WriteUnits = 2;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.getWriteunits = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.setWriteunits = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional int64 StorageSize = 3;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.getStoragesize = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 3, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.setStoragesize = function(value) {
  return jspb.Message.setProto3IntField(this, 3, value);
};


/**
 * optional int64 ReadUnitsThrottled = 4;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.getReadunitsthrottled = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 4, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.setReadunitsthrottled = function(value) {
  return jspb.Message.setProto3IntField(this, 4, value);
};


/**
 * optional int64 WriteUnitsThrottled = 5;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.getWriteunitsthrottled = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 5, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.setWriteunitsthrottled = function(value) {
  return jspb.Message.setProto3IntField(this, 5, value);
};


/**
 * optional int64 StorageSizeThrottled = 6;
 * @return {number}
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.getStoragesizethrottled = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 6, 0));
};


/**
 * @param {number} value
 * @return {!proto.tigrisdata.observability.v1.QuotaUsageResponse} returns this
 */
proto.tigrisdata.observability.v1.QuotaUsageResponse.prototype.setStoragesizethrottled = function(value) {
  return jspb.Message.setProto3IntField(this, 6, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.GetInfoRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.GetInfoRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.GetInfoRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.GetInfoRequest.toObject = function(includeInstance, msg) {
  var f, obj = {

  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.GetInfoRequest}
 */
proto.tigrisdata.observability.v1.GetInfoRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.GetInfoRequest;
  return proto.tigrisdata.observability.v1.GetInfoRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.GetInfoRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.GetInfoRequest}
 */
proto.tigrisdata.observability.v1.GetInfoRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.GetInfoRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.GetInfoRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.GetInfoRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.GetInfoRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.tigrisdata.observability.v1.GetInfoResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.tigrisdata.observability.v1.GetInfoResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.tigrisdata.observability.v1.GetInfoResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.GetInfoResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    serverVersion: jspb.Message.getFieldWithDefault(msg, 1, ""),
    error: (f = msg.getError()) && proto.tigrisdata.observability.v1.Error.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.tigrisdata.observability.v1.GetInfoResponse}
 */
proto.tigrisdata.observability.v1.GetInfoResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.tigrisdata.observability.v1.GetInfoResponse;
  return proto.tigrisdata.observability.v1.GetInfoResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.tigrisdata.observability.v1.GetInfoResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.tigrisdata.observability.v1.GetInfoResponse}
 */
proto.tigrisdata.observability.v1.GetInfoResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setServerVersion(value);
      break;
    case 2:
      var value = new proto.tigrisdata.observability.v1.Error;
      reader.readMessage(value,proto.tigrisdata.observability.v1.Error.deserializeBinaryFromReader);
      msg.setError(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.tigrisdata.observability.v1.GetInfoResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.tigrisdata.observability.v1.GetInfoResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.tigrisdata.observability.v1.GetInfoResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.tigrisdata.observability.v1.GetInfoResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getServerVersion();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getError();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      proto.tigrisdata.observability.v1.Error.serializeBinaryToWriter
    );
  }
};


/**
 * optional string server_version = 1;
 * @return {string}
 */
proto.tigrisdata.observability.v1.GetInfoResponse.prototype.getServerVersion = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.tigrisdata.observability.v1.GetInfoResponse} returns this
 */
proto.tigrisdata.observability.v1.GetInfoResponse.prototype.setServerVersion = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional Error error = 2;
 * @return {?proto.tigrisdata.observability.v1.Error}
 */
proto.tigrisdata.observability.v1.GetInfoResponse.prototype.getError = function() {
  return /** @type{?proto.tigrisdata.observability.v1.Error} */ (
    jspb.Message.getWrapperField(this, proto.tigrisdata.observability.v1.Error, 2));
};


/**
 * @param {?proto.tigrisdata.observability.v1.Error|undefined} value
 * @return {!proto.tigrisdata.observability.v1.GetInfoResponse} returns this
*/
proto.tigrisdata.observability.v1.GetInfoResponse.prototype.setError = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.tigrisdata.observability.v1.GetInfoResponse} returns this
 */
proto.tigrisdata.observability.v1.GetInfoResponse.prototype.clearError = function() {
  return this.setError(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.tigrisdata.observability.v1.GetInfoResponse.prototype.hasError = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * @enum {number}
 */
proto.tigrisdata.observability.v1.Code = {
  OK: 0,
  CANCELLED: 1,
  UNKNOWN: 2,
  INVALID_ARGUMENT: 3,
  DEADLINE_EXCEEDED: 4,
  NOT_FOUND: 5,
  ALREADY_EXISTS: 6,
  PERMISSION_DENIED: 7,
  RESOURCE_EXHAUSTED: 8,
  FAILED_PRECONDITION: 9,
  ABORTED: 10,
  OUT_OF_RANGE: 11,
  UNIMPLEMENTED: 12,
  INTERNAL: 13,
  UNAVAILABLE: 14,
  DATA_LOSS: 15,
  UNAUTHENTICATED: 16,
  CONFLICT: 17,
  BAD_GATEWAY: 18,
  METHOD_NOT_ALLOWED: 19,
  CONTENT_TOO_LARGE: 20
};

/**
 * @enum {number}
 */
proto.tigrisdata.observability.v1.TigrisOperation = {
  ALL: 0,
  READ: 1,
  WRITE: 2,
  METADATA: 3
};

/**
 * @enum {number}
 */
proto.tigrisdata.observability.v1.MetricQueryFunction = {
  RATE: 0,
  COUNT: 1,
  NONE: 2
};

/**
 * @enum {number}
 */
proto.tigrisdata.observability.v1.RollupAggregator = {
  ROLLUP_AGGREGATOR_SUM: 0,
  ROLLUP_AGGREGATOR_COUNT: 1,
  ROLLUP_AGGREGATOR_MIN: 2,
  ROLLUP_AGGREGATOR_MAX: 3,
  ROLLUP_AGGREGATOR_AVG: 4
};

/**
 * @enum {number}
 */
proto.tigrisdata.observability.v1.MetricQuerySpaceAggregation = {
  AVG: 0,
  MIN: 1,
  MAX: 2,
  SUM: 3
};

goog.object.extend(exports, proto.tigrisdata.observability.v1);
