// @tigrisdata/core/dist/utils/logger.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Log = void 0;
const chalk_1 = __importDefault(require("chalk"));
class Logger {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    constructor() {
        this.prefixes = {
            debug: chalk_1.default.green("debug") + " -",
            info: chalk_1.default.cyan("info") + " -",
            warn: chalk_1.default.yellow("warn") + " -",
            error: chalk_1.default.red("error") + " -",
            event: chalk_1.default.magenta("event") + " -",
        };
    }
    static get Instance() {
        if (!Logger._instance) {
            Logger._instance = new Logger();
        }
        return this._instance;
    }
    debug(...message) {
        console.log(this.prefixes.debug, ...message);
    }
    info(...message) {
        console.log(this.prefixes.info, ...message);
    }
    error(...message) {
        console.log(this.prefixes.error, ...message);
    }
    warn(...message) {
        console.log(this.prefixes.warn, ...message);
    }
    event(...message) {
        console.log(this.prefixes.event, ...message);
    }
}
exports.Log = Logger.Instance;
