// @getmetal/metal-sdk/dist/src/motorhead.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
const axios_1 = __importDefault(require("axios"));
const constants_1 = require("./constants");
const MANAGED_BASE_URL = `${constants_1.API_URL}/v1/motorhead`;
class Motorhead {
    constructor({ apiKey, clientId, baseUrl = MANAGED_BASE_URL }) {
        const isManaged = baseUrl === MANAGED_BASE_URL;
        if (isManaged && (!apiKey || !clientId)) {
            throw new Error('apiKey and clientId required for managed motorhead');
        }
        this.apiKey = apiKey;
        this.clientId = clientId;
        this.baseUrl = baseUrl;
    }
    addMemory(sessionId, payload) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const { data } = yield axios_1.default.post(`${this.baseUrl}/sessions/${sessionId}/memory`, payload, {
                headers: {
                    'Content-Type': 'application/json',
                    'x-metal-api-key': this.apiKey,
                    'x-metal-client-id': this.clientId,
                },
            });
            const memory = (_a = data === null || data === void 0 ? void 0 : data.data) !== null && _a !== void 0 ? _a : data;
            return memory;
        });
    }
    getMemory(sessionId) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const { data } = yield axios_1.default.get(`${this.baseUrl}/sessions/${sessionId}/memory`, {
                headers: {
                    'Content-Type': 'application/json',
                    'x-metal-api-key': this.apiKey,
                    'x-metal-client-id': this.clientId,
                },
            });
            const memory = (_a = data === null || data === void 0 ? void 0 : data.data) !== null && _a !== void 0 ? _a : data;
            return memory;
        });
    }
    deleteMemory(sessionId) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const { data } = yield axios_1.default.delete(`${this.baseUrl}/sessions/${sessionId}/memory`, {
                headers: {
                    'Content-Type': 'application/json',
                    'x-metal-api-key': this.apiKey,
                    'x-metal-client-id': this.clientId,
                },
            });
            return (_a = data === null || data === void 0 ? void 0 : data.data) !== null && _a !== void 0 ? _a : data;
        });
    }
}
module.exports = Motorhead;
