// @tigrisdata/core/dist/search/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocStatus = exports.DeleteIndexResponse = exports.IndexInfo = void 0;
const utility_1 = require("../utility");
const constants_1 = require("../constants");
const error_1 = require("../error");
class IndexInfo {
    constructor(name, schema) {
        this._name = name;
        this._schema = schema;
    }
    static from(info) {
        const schema = info.getSchema()
            ? JSON.parse(utility_1.Utility._base64Decode(info.getSchema_asB64()))
            : {};
        return new this(info.getName(), schema);
    }
    get name() {
        return this._name;
    }
    get schema() {
        return this._schema;
    }
}
exports.IndexInfo = IndexInfo;
class DeleteIndexResponse {
    constructor(message) {
        this.status = constants_1.Status.Deleted;
        this._message = message;
    }
    get message() {
        return this._message;
    }
    static from(resp) {
        return new this(resp.getMessage());
    }
}
exports.DeleteIndexResponse = DeleteIndexResponse;
class DocStatus {
    constructor(id, error) {
        this.id = id;
        this.error = error;
    }
    static from(protoStatus) {
        const err = protoStatus.hasError()
            ? new error_1.TigrisError(protoStatus.getError().getMessage())
            : undefined;
        return new this(protoStatus.getId(), err);
    }
}
exports.DocStatus = DocStatus;
