// @gomomento/sdk/dist/src/config/middleware/experimental-request-logging-middleware.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExperimentalRequestLoggingMiddleware = void 0;
class ExperimentalLoggingMiddlewareRequestHandler {
    constructor(logger, requestId) {
        this.logger = logger;
        this.requestId = requestId;
    }
    onRequestMetadata(metadata) {
        this.logger.debug('Logging middleware: request %s onRequestMetadata: %s', this.requestId, JSON.stringify(metadata.toJSON()));
        return Promise.resolve(metadata);
    }
    onRequestBody(request) {
        this.logger.debug('Logging middleware: request %s onRequestBody: request type: %s, request size: %s', this.requestId, 
        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
        request.constructor.name, request.serializeBinary().length);
        return Promise.resolve(request);
    }
    onResponseMetadata(metadata) {
        this.logger.debug('Logging middleware: request %s onResponseMetadata: %s', this.requestId, JSON.stringify(metadata.toJSON()));
        return Promise.resolve(metadata);
    }
    onResponseBody(response) {
        var _a, _b, _c, _d;
        this.logger.debug('Logging middleware: request %s onResponseBody: response type: %s, response size: %s', this.requestId, 
        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
        (_b = (_a = response === null || response === void 0 ? void 0 : response.constructor) === null || _a === void 0 ? void 0 : _a.name) !== null && _b !== void 0 ? _b : 'null', (_d = (_c = response === null || response === void 0 ? void 0 : response.serializeBinary()) === null || _c === void 0 ? void 0 : _c.length) !== null && _d !== void 0 ? _d : 0);
        return Promise.resolve(response);
    }
    onResponseStatus(status) {
        this.logger.debug('Logging middleware: request %s onResponseStatus: status code: %s', this.requestId, 
        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
        status.code);
        return Promise.resolve(status);
    }
}
/**
 * This middleware implements per-request logging which can be used for
 * debugging.  The log format is currently considered experimental; in a
 * future release, once the log format is considered stable, this class will
 * be renamed to remove the Experimental prefix.
 *
 * WARNING: enabling this middleware may have minor performance implications,
 * so enable with caution.
 *
 * In order for this middleware to produce output you will need to have
 * set up your {Configuration} with a {MomentoLoggerFactory} instance that
 * is configured to log at DEBUG level or lower.  See `advanced.ts` in the
 * examples directory for an example of how to set up your {Configuration} to
 * enable this middleware.
 */
class ExperimentalRequestLoggingMiddleware {
    constructor(loggerFactory) {
        this.logger = loggerFactory.getLogger(this);
        this.nextRequestId = 0;
    }
    onNewRequest() {
        this.nextRequestId++;
        return new ExperimentalLoggingMiddlewareRequestHandler(this.logger, this.nextRequestId.toString());
    }
}
exports.ExperimentalRequestLoggingMiddleware = ExperimentalRequestLoggingMiddleware;
