// @gomomento/sdk-core/dist/src/messages/responses/cache-sorted-set-get-score.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Error = exports.Miss = exports.Hit = exports.Response = void 0;
const response_base_1 = require("./response-base");
/**
 * Parent response type for a sorted set get score request.  The
 * response object is resolved to a type-safe object of one of
 * the following subtypes:
 *
 * - {Hit}
 * - {Miss}
 * - {Error}
 *
 * `instanceof` type guards can be used to operate on the appropriate subtype.
 * @example
 * For example:
 * ```
 * if (response instanceof CacheSortedSetGetScore.Error) {
 *   // Handle error as appropriate.  The compiler will smart-cast `response` to type
 *   // `CacheSortedSetGetScore.Error` in this block, so you will have access to the properties
 *   // of the Error class; e.g. `response.errorCode()`.
 * }
 * ```
 */
class Response extends response_base_1.ResponseBase {
}
exports.Response = Response;
class _Hit extends Response {
    constructor(score, value) {
        super();
        this._value = value;
        this._score = score;
    }
    /**
     * Returns the value of the element in the sorted set as a Uint8Array.
     * @returns {number}
     */
    valueUint8Array() {
        return this._value;
    }
    /**
     * Returns the score of the element in the sorted set as a string.
     * @returns {string}
     */
    valueString() {
        return new TextDecoder().decode(this._value);
    }
    /**
     *
     *
     * @return {*}  {number}
     * @memberof _Hit
     */
    score() {
        return this._score;
    }
    toString() {
        return `${super.toString()}: value: ${this.valueString()}, score: ${this.score()}`;
    }
}
/**
 * Indicates that the requested data was successfully retrieved from the cache.  Provides
 * `score()` accessor to retrieve score.
 */
class Hit extends (0, response_base_1.ResponseHit)(_Hit) {
}
exports.Hit = Hit;
class _Miss extends Response {
    constructor(value) {
        super();
        this._value = value;
    }
    /**
     * Returns the value of the element in the sorted set as a Uint8Array.
     * @returns {number}
     */
    valueUint8Array() {
        return this._value;
    }
    /**
     * Returns the score of the element in the sorted set as a string.
     * @returns {string}
     */
    valueString() {
        return new TextDecoder().decode(this._value);
    }
    toString() {
        return `${super.toString()}: value: ${this.valueString()}`;
    }
}
/**
 * Indicates that the requested data was not available in the cache.
 */
class Miss extends (0, response_base_1.ResponseMiss)(_Miss) {
}
exports.Miss = Miss;
class _Error extends Response {
    constructor(_innerException, value) {
        super();
        this._innerException = _innerException;
        this._value = value;
    }
    /**
     * Returns the value of the element in the sorted set as a Uint8Array.
     * @returns {number}
     */
    valueUint8Array() {
        return this._value;
    }
    /**
     * Returns the score of the element in the sorted set as a string.
     * @returns {string}
     */
    valueString() {
        return new TextDecoder().decode(this._value);
    }
}
/**
 * Indicates that an error occurred during the sorted set get score request.
 *
 * This response object includes the following fields that you can use to determine
 * how you would like to handle the error:
 *
 * - `errorCode()` - a unique Momento error code indicating the type of error that occurred.
 * - `message()` - a human-readable description of the error
 * - `innerException()` - the original error that caused the failure; can be re-thrown.
 */
class Error extends (0, response_base_1.ResponseError)(_Error) {
}
exports.Error = Error;
