// chromadb/dist/module/generated/models.js
/* eslint-disable */
// tslint:disable
/**
 * FastAPI
 *
 *
 * OpenAPI spec version: 0.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator+.
 * https://github.com/karlvr/openapi-generator-plus
 * Do not edit the class manually.
 */
export var Api;
(function (Api) {
    /**
     * @export
     * @namespace GetEmbedding
     */
    let GetEmbedding;
    (function (GetEmbedding) {
        let IncludeEnum;
        (function (IncludeEnum) {
            IncludeEnum["Documents"] = "documents";
            IncludeEnum["Embeddings"] = "embeddings";
            IncludeEnum["Metadatas"] = "metadatas";
            IncludeEnum["Distances"] = "distances";
        })(IncludeEnum = GetEmbedding.IncludeEnum || (GetEmbedding.IncludeEnum = {}));
    })(GetEmbedding = Api.GetEmbedding || (Api.GetEmbedding = {}));
    /**
     * @export
     * @namespace QueryEmbedding
     */
    let QueryEmbedding;
    (function (QueryEmbedding) {
        let IncludeEnum;
        (function (IncludeEnum) {
            IncludeEnum["Documents"] = "documents";
            IncludeEnum["Embeddings"] = "embeddings";
            IncludeEnum["Metadatas"] = "metadatas";
            IncludeEnum["Distances"] = "distances";
        })(IncludeEnum = QueryEmbedding.IncludeEnum || (QueryEmbedding.IncludeEnum = {}));
    })(QueryEmbedding = Api.QueryEmbedding || (Api.QueryEmbedding = {}));
})(Api || (Api = {}));
