// @gomomento/sdk-core/dist/src/auth/tokens/token-scope.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PredefinedScope = exports.AllDataReadWrite = exports.Permissions = exports.TopicPermission = exports.TopicRole = exports.CachePermission = exports.CacheRole = void 0;
var CacheRole;
(function (CacheRole) {
    CacheRole["None"] = "none";
    CacheRole["ReadWrite"] = "readwrite";
})(CacheRole = exports.CacheRole || (exports.CacheRole = {}));
class CachePermission {
    constructor(cacheRole) {
        this.cacheRole = cacheRole;
    }
}
exports.CachePermission = CachePermission;
var TopicRole;
(function (TopicRole) {
    TopicRole["None"] = "none";
    TopicRole["ReadWrite"] = "readwrite";
})(TopicRole = exports.TopicRole || (exports.TopicRole = {}));
class TopicPermission {
    constructor(topicRole) {
        this.topicRole = topicRole;
    }
}
exports.TopicPermission = TopicPermission;
class Permissions {
    constructor(permissions) {
        this.permissions = permissions;
    }
}
exports.Permissions = Permissions;
exports.AllDataReadWrite = new Permissions([
    new CachePermission(CacheRole.ReadWrite),
    new TopicPermission(TopicRole.ReadWrite),
]);
class PredefinedScope {
}
exports.PredefinedScope = PredefinedScope;
